Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -365,55 +368,172 @@ def start_chrome_headless(chrome_port, chrome_binary=CHROME_EXE, user_options=""
-    browser = None
-    chrome_port = read_info_file(PORT_FILE_PATH)
-    if chrome_port:
-        browser = is_chrome_running_locally(chrome_port)
-        if browser:
-            return browser, chrome_port
+    if not instance_id:
+        instance_id = 'None'
+    if not chrome_options:
+        chrome_options = 'None'
+
+    chrome_instances_contents = read_file(CHROME_INSTANCES_FILE_PATH)
+
+    instance_id_to_chrome_options, chrome_options_to_port, instance_id_to_port = get_chrome_instances_contents_dictionaries(
+        chrome_instances_contents)
+
+    if chrome_instances_contents:
+        splitted_chrome_instances_contents = chrome_instances_contents.strip().splitlines()
+        _, instances_id_tuple, chromes_options_tuple = zip(
+            *[line.strip().split('\t') for line in splitted_chrome_instances_contents])
+        instances_id = list(instances_id_tuple) if instances_id_tuple else []
+        chromes_options = list(chromes_options_tuple) if chromes_options_tuple else []
+
+        if instances_id and not chromes_options:
+            chromes_options.append('None')
+
+        demisto.debug(f"Chrome instances contents: {chromes_options=}, {instances_id=}")
+
+    if not chrome_instances_contents or (chrome_options in chromes_options and instance_id not in instances_id) or (
+        chrome_options not in chromes_options and instance_id not in instances_id):
+        """
+        When:
+            case 1: file is empty, that means no chrome open on the machine
+            or
+            case 2: new instance with the same existing chrome configuration as another instance
+            or
+            case 3: chrome configuration does not exist and integration instance does not exist in the file
+        Then:
+            create new browser
+        """
+
+        if not chrome_instances_contents:
+            demisto.debug("case 1: file is empty, that means no chrome open on the machine")
         else:
-            write_info_file(PORT_FILE_PATH, '')
+            if chrome_options in chromes_options and instance_id not in instances_id:
+                demisto.debug("case 2: new instance with the same existing chrome configuration as another instance")
+            elif chrome_options not in chromes_options and instance_id not in instances_id:
+                demisto.debug("case 3: chrome configuration does not exist and integration instance does not exist in the file")
+
+        return generate_new_chrome_instance(instance_id, chrome_options)
+
+    elif instance_id in instances_id and chrome_options != instance_id_to_chrome_options.get(instance_id):
+        """
+        When:
+            case 4: instance updated the chrome configuration to different configuration
+        Then:
+            should kill the old browser and create new one
+        """
+
+        demisto.debug(f"case 4: found {instance_id=} but with different {chrome_options=}")
+        chrome_port = instance_id_to_port.get(instance_id)
```

Comment:
What if there's no chrome running for that instance?
In that case, `chrome_port` will be empty.
Can the code handle this scenario?

Reply:
if there's no chrome running for that instance, it can get that case.
the entrance of that code is if instance_id in instances_id, means instance have already chrome open.
if, for some reason, in some instance it open chrome and wrote the port in the chrome_instances.tsv file and somehow the chrome closed. should i deal with that? i think that it covered later on when the browser that returned will be None..

Reply:
@ilaredo the first part of your comment is correct - according to your implementation it should not occur.

what you raised in the second part of your comment is a different topic (and is not really related to your task, as it can allegedly happen today without your changes) - what is the possibility that a chrome browser will be unexpectedly closed. @tkatzir will probably know the answer better than me, but anyway we don't need to address it on this PR.